src/io/DumpXYZ.cpp

```diff
@@ -34,47 +38,151 @@ using namespace std;
-          if(length_factor == 1.0){
+          std::streamsize p = myfile.precision();
+          if(length_factor == 1.0){ // probably can remove this cond branch and
+            // just keep the second one where len_fact != 1.0
+            /*
             for(size_t i=0; i<num_of_particles; i++){
             	//myfile << "bla" << endl;
               myfile << "  0  " << cei.nextProperties() << endl;
+              //myfile << currentId() << " " << cei.nextProperties() << endl;
               //myfile << "  0  " << cei.nextCoordinates() << endl;
             }
+            */
+
+            for (size_t i=0; i<num_of_particles; i++) {
+
+              // // print as fastwritexyz in Py, given number (10) of double digits
+
+
+              if (getStorePids()) {
+
+                // resemble fastwritexyz() which stores or not the pids
+                // (non XYZ standard, but fastwritexyz does that)
+                // precision for doubles set as the fastwritexyz() python func
+                if (getStoreVelocities() == false) {
+                  myfile << cei.currentId() << " " << cei.currentProperties()[0] <<
+                  " " << fixed << setprecision(10) << cei.currentProperties()[1] << " " << cei.currentProperties()[2] <<
+                  " " << cei.currentProperties()[3] << endl;
+                  myfile.unsetf(ios_base::fixed);
+                  myfile << setprecision(p);
+                } else {
+
+                  myfile << cei.currentId() << " " << cei.currentProperties()[0] <<
+                  " " << fixed << setprecision(10) << cei.currentProperties()[1] << " " << cei.currentProperties()[2] <<
+                  " " << cei.currentProperties()[3] << " " << cei.currentProperties()[4] << " " << cei.currentProperties()[5] <<
+                  " " << cei.currentProperties()[6] << endl;
+                  myfile.unsetf(ios_base::fixed);
+                  myfile << setprecision(p);
+
+                }
+
+              } else { // standard XYZ, no PID at the beginning of the line
+                if (getStoreVelocities() == false) {
+                  myfile << cei.currentProperties()[0] <<
+                  " " << fixed << setprecision(10) << cei.currentProperties()[1] << " " << cei.currentProperties()[2] <<
+                  " " << cei.currentProperties()[3] << endl;
+                  myfile.unsetf(ios_base::fixed);
+                  myfile << setprecision(p);
+
+                } else {
+
+                  myfile << cei.currentProperties()[0] <<
+                  " " << fixed << setprecision(10) << cei.currentProperties()[1] << " " << cei.currentProperties()[2] <<
+                  " " << cei.currentProperties()[3] << " " << cei.currentProperties()[4] << " " << cei.currentProperties()[5] <<
+                  " " << cei.currentProperties()[6] << endl;
+                  myfile.unsetf(ios_base::fixed);
+                  myfile << setprecision(p);
+
```

Comment:
Wouldn't it be cleaner to append to the current line  if `getStoreVelocities` or `getStoreVelocities` is set instead of duplicating everything?